wave_matching/src/icp.cpp

```diff
@@ -31,6 +31,18 @@ ICPMatcher::ICPMatcher(float res) {
     this->icp.setEuclideanFitnessEpsilon(fit_eps);
 }
 
+ICPMatcher::~ICPMatcher() {
+    if (this->ref) {
+        this->ref.reset();
+    }
+    if (this->target) {
+        this->target.reset();
+    }
+    if (this->final) {
+        this->final.reset();
+    }
+}
```

Comment:
Are these resets actally needed? won't they reset themselves when they go out of scope? Just a question.